framework/auth/core.py

```diff
@@ -1361,6 +1369,13 @@ def has_inst_auth(self, inst):
             return True
         return False
 
+    def remove_inst(self, inst_id):
+        inst = Institution.load(inst_id)
```

Comment:
Is it necessary to load the whole `Institution` record here? I believe you can pass a PK to `remove`, but you should try it out.